spec/Section 6 -- Execution.md

```diff
@@ -26,11 +26,15 @@ request is determined by the result of executing this operation according to the
 "Executing Operations” section below.
 
 ExecuteRequest(schema, document, operationName, variableValues, initialValue):
-
+  Note: the execution assumes implementing language supports coroutines.
+  Alternatively, the socket can provide a write buffer pointer to allow {ExecuteRequest()}
+  to directly write payloads into the buffer. 
   * Let {operation} be the result of {GetOperation(document, operationName)}.
   * Let {coercedVariableValues} be the result of {CoerceVariableValues(schema, operation, variableValues)}.
   * If {operation} is a query operation:
-    * Return {ExecuteQuery(operation, schema, coercedVariableValues, initialValue)}.
+    * Let {payloads} be the iterator obtained from calling {ExecuteQuery(operation, schema, coercedVariableValues, initialValue, subsequentPayloads)}.
+    * For each {payload} in {payloads}:
+      * Yield {payload}.
```

Comment:
This isn't guaranteed to be an iterator (`ExecuteQuery` may return an unordered map):

```suggestion
    * Let {executionResult} be the result of calling {ExecuteQuery(operation, schema, coercedVariableValues, initialValue, subsequentPayloads)}.
    * If {executionResult} is an iterator:
      * For each {payload} in {executionResult}:
        * Yield {payload}.
    * Otherwise:
      * Return {executionResult}.
```

I'm also not sure if we need an explicit return once the `For each` exits.

Further, with coroutines, are we explicitly sending the termination signal to the iterator? E.g. if the iterator is sending 1,000,000 results, and the client closes the connection after 20,000, do we need to explicitly state that this should terminate the iterator?

Reply:
I added details about handling termination signals in https://github.com/graphql/graphql-spec/pull/742/commits/aa2741dcca1253e4741c4924a36399b6ba10139a